### PR TITLE
Use LazyUser when creating the user folder

### PR DESCRIPTION
**Pull request by nickvergessen**
Meme channel query count:
Before | After
---|---
86 | 69

### CODE REVIEW

File: lib/private/Files/Node/Root.php [hunk 1]
```diff
@@ -477,7 +478,7 @@ public function getByIdInPath(int $id, string $path): array {
 			$absolutePath = rtrim($mount->getMountPoint() . $pathRelativeToMount, '/');
 			return $this->createNode($absolutePath, new FileInfo(
 				$absolutePath, $mount->getStorage(), $cacheEntry->getPath(), $cacheEntry, $mount,
-				\OC::$server->getUserManager()->get($mount->getStorage()->getOwner($pathRelativeToMount))
+				new LazyUser($mount->getStorage()->getOwner($pathRelativeToMount), \OC::$server->getUserManager())
```
[the remark this answers is not in the document]
Comment (nickvergessen) [reply lib/private/Files/Node/Root.php]: Fixed

**Review by CarlSchwan — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]

**Comment by PVince81** [comment]
psalm seems unhappy

File: lib/private/Files/Node/Root.php [hunk 2]
```diff
@@ -362,7 +363,7 @@ public function getName() {
 	 * @throws NotPermittedException
 	 */
 	public function getUserFolder($userId) {
-		$userObject = $this->userManager->get($userId);
+		$userObject = new LazyUser($userId, $this->userManager);
 
 		if (is_null($userObject)) {
```
Comment (nickvergessen) [inline lib/private/Files/Node/Root.php:368]: Can someone confirm it's save to remove this full block?

Comment (CarlSchwan) [inline lib/private/Files/Node/Root.php:368]:
```suggestion
		if (is_null($userObject->getDisplayName())) {
```

Comment (CarlSchwan) [reply lib/private/Files/Node/Root.php:368]: Oh wait that won't work 😩

**Comment by nickvergessen** [comment]
Closing for now as we still see the error in our log from time to time which would then be unreachable

**Pull request closed without merging**